internal/indexer/indexer.go

```diff
@@ -142,6 +144,39 @@ func (i *Indexer) loadPackages() error {
 		i.packages = pkgs
 		i.packagesByFile = map[string][]*packages.Package{}
 
+		shouldVisitPackage := func(p *packages.Package) bool {
+			// The loader returns 4 packages because (loader.Config).Tests==true and we
+			// want to avoid duplication.
+			if p.Name == "main" && strings.HasSuffix(p.ID, ".test") {
+				return false // synthesized `go test` program
+			}
+			if strings.HasSuffix(p.Name, "_test") {
+				return true
+			}
+
+			// Index only the combined test package if it's present. If the package has no test files,
+			// it won't be present, and we need to just index the default package.
```

Comment:
@efritz @beyang can either of you explain to me what "index only the combined test package if it's present." means? I am specifically confused how two packages can be _combined_? Is this an LSIF concept, or a `go/loader` concept I'm not aware of?

Reply:
Figured it out: https://github.com/sourcegraph/lsif-go/pull/138/files#diff-87c2475baa87005bd3a3001ec33f201feea23caddbbf5a2f249238a2bd32026dR158-R170